astropath/scripts/use_catalogs.py

```diff
@@ -0,0 +1,156 @@
+        query_fields = ['rPSFLikelihood']
+    elif pargs.survey == 'DECaL':
+        query_fields = ['shapedev_r', 'shapeexp_r']
+
+    # Grab the catalo
+    catalog = survey.get_catalog(query_fields=query_fields)
+
+    if len(catalog) == 0:
+        print(f"No objects in the catalog of your survey within {pargs.ssize} arcmin")
+        return
+
+    # Clean up the catalog
+    if pargs.survey == 'Pan-STARRS':
+        cut_size = catalog['rKronRad'] > 0.
+        cut_mag = catalog['Pan-STARRS_r'] > 14. # Reconsider this
+        cut_point = np.log10(np.abs(catalog['rPSFLikelihood'])) < (-2)
+        keep = cut_size & cut_mag & cut_point
+        # Half-light radius
+        mag_key = 'Pan-STARRS_r'
+        catalog['ang_size'] = catalog['rKronRad'].copy() 
+    elif pargs.survey == 'DECaL':
+        mag_key = 'DECaL_r'
+        # Cuts
+        cut_mag = (catalog[mag_key] > 14.) & np.isfinite(catalog[mag_key]) 
+        cut_star = catalog['gaia_pointsource'] == 0
+        keep = cut_mag & cut_star
+        # Half-light radius
+        catalog['ang_size'] = np.maximum(catalog['shapedev_r'], catalog['shapeexp_r'])
+        zero = catalog['ang_size'] == 0.
+        catalog['ang_size'][zero] = 1. # KLUDGE!!
+    else:
+        raise IOError(f"Not ready for this survey: {pargs.survey}")
+
+
+
+    catalog = catalog[keep]
+
+    if pargs.debug:
+        if pargs.survey == 'Pan-STARRS':
+            sns.histplot(x=catalog['Pan-STARRS_r'])#, bins=20)
+            plt.show()
+            sns.histplot(x=catalog['rKronRad'])#, bins=20)
+            plt.show()
+            sns.histplot(x=catalog['rPSFLikelihood'])#, bins=20)
+            plt.show()
+            embed(header='lowdm_bb: Need to set boxsize')
+
+    # Set boxsize accoring to the largest galaxy (arcsec)
+    box_hwidth = max(30., 10.*np.max(catalog['ang_size']))
+
+    # Turn into a cndidate table
+    Path = path.PATH()
+
+   # Set up localization
+    Path.init_localization('eellipse', 
+                           center_coord=coord, 
+                           eellipse=eellipse)
+    # Coords
+    Path.init_candidates(catalog['ra'],
+                         catalog['dec'],
+                         catalog['ang_size'],
+                         mag=catalog[mag_key])
+
+    # Candidate prior
+    Path.init_cand_prior('inverse', P_U=pargs.PU)
+
+    # Offset prior
+    Path.init_theta_prior('exp', 6., scale)
+
+    # Priors
+    p_O = Path.calc_priors()
+
+    # Posterior
+    P_Ox, P_Ux = Path.calc_posteriors('local', box_hwidth=box_hwidth, 
+        max_radius=box_hwidth)
+
+    # Print
+    Path.candidates.sort_values(by='P_Ox', ascending=False, inplace=True)
+    print(Path.candidates[['ra', 'dec', 'ang_size', 'mag', 'P_O', 'P_Ox']])
+    print(f"P_Ux = {Path.candidates['P_Ux'][0]}")
+
+    # Save?
+    if pargs.outfile is not None:
+        Path.candidates.to_csv(pargs.outfile, index=False)
+        print(f"Wrote: {pargs.outfile}")
+
+    # Return
+    return Path
+
+# Test
```

Comment:
Can this documentation be made to show with the parser help text?

Reply:
instead, I generated this:

https://github.com/FRBs/astropath/blob/auto_path_with_catalog-/docs/scripts.rst